firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -286,19 +294,23 @@ public void deleteFirebaseInstallation(
       httpURLConnection.addRequestProperty("Authorization", "FIS_v2 " + refreshToken);
 
       int httpResponseCode = httpURLConnection.getResponseCode();
-      httpURLConnection.disconnect();
 
       if (httpResponseCode == 200 || httpResponseCode == 401 || httpResponseCode == 404) {
         return;
       }
 
+      logFisCommunicationError(httpURLConnection);
+      httpURLConnection.disconnect();
```

Comment:
If you return before this httpURLConnection seems to stay connected, because you only call #disconnect over here.

Reply:
Good catch. Fixed it.